### DISCUSSION TITLE
Exporting System metrics using OpenTelemetry Java Auto Instrumentation

### DISCUSSION THREAD

**Discussion by shravani154** [open-telemetry/opentelemetry-java-instrumentation#6349] [Q&A]
I have an application running in kubernetes which is instrumented with open telemetry java agent. I am exporting metrics to otlp collector where I am unable to see system metrics like cpu usage. Is there any external configuration that I can do to export system metrics?

Thanks in advance

**Comment by mateuszrzeszutek** [comment]
Hey @shravani154 ,
If you're using the collector in [the agent mode](https://opentelemetry.io/docs/collector/deployment/#agent) then you could configure the collector to gather the system metrics for each pod (e.g. using the [host metrics receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/hostmetricsreceiver)).
Alternatively, the agent contains an optional Oshi instrumentation that will emit some system metrics (less than what you'd get from the collector though) - unfortunately its usage is not documented (#1566), but pretty much all you need to do is to add a jar containing the Oshi library to the classpath and it should work.

**Reply by shravani154** [reply]
Thank you for the suggestions.
I have tried using oshi instrumentation but did not receive any system metrics like system memory usage and system memory utilization. 
I was able to see these metrics when the application is running in local machine. The issue is when running it in a kubernetes pod. let me know if there is external configuration to be done for kubernetes pods.
